src/zoid/buttons/util.js

```diff
@@ -74,3 +74,35 @@ export function getVenmoExperiment(experiment : ?Experiment) : VenmoExperiment {
         enableVenmo: Boolean(isExperimentEnabled || isEnableFundingVenmo)
     };
 }
+
+export function applePaySession() : Function {
+    try {
+        if (!window.ApplePaySession) {
+            return undefined;
+        }
+
+        return (version, request) => {
+            const session = new window.ApplePaySession(version, request);
+            return {
+                begin: session.begin(),
+                on:    (name, handler) => {
+                    const validNames = [
+                        'validateMerchant',
+                        'paymentmethodselected',
+                        'shippingmethodselected',
+                        'shippingcontactselected',
+                        'paymentauthorized',
+                        'cancel'
+                    ];
+                    if (validNames.indexOf(name) === -1) {
+                        // eslint-disable-next-line no-console
+                        console.error(`Invalid ApplePaySession event ${ name }`);
+                    }
+                    session[`on${ name }`] = handler;
```

Comment:
Does applepay not support `addEventListener`?

Reply:
Just thought it was a bit long.